pennylane_ionq/api_client.py

```diff
@@ -350,6 +352,10 @@ def refresh_data(self, data):
         for field in self.resource.fields:
             field.set(data.get(field.name, None))
 
+        if "results_url" in data.keys():
+            result = self.client.get(self.join_path(data["results_url"]), params=params)
+            self.resource.fields[-1].set(result.json())
```

Comment:
I suppose `self.resource` is a `Job` type, and `.fields[-1]` is the `data` field? if so, can we ensure this more robustly, or at least leave a comment in the `Job` field definitions ensuring that `data` remains last?

Reply:
Good point - I'm not sure how to do it more robustly without changing the format of what is returned for `fields` (which would maybe be fine, but I'm trying to change as little as possible), but I added a comment in the fields definition